src/DonationForms/Properties/FormSettings.php

```diff
@@ -234,10 +234,9 @@ public static function fromArray(array $array): self
      */
     public static function fromJson(string $json): self
     {
-        $self = new self();
-        $array = json_decode($json, true);
-
-        return $self::fromArray($array);
+        return self::fromArray(
+            json_decode($json, true)
+        );
```

Comment:
Cool, I just realized `json_decode('', true);` or any invalid json like `json_decode('{"hello": "\\hello"}', true);` returns null.  I think to be safe, we should validate the string as json, an array or just cast the json_decode as an array so it falls back to the default FormSettings object.

Reply:
My concern is that falling back would be a silent failure.

That said, I think we can get this merged and revisit this part.

Reply:
I'm just concerned that if the settings are corrupt it will break the form and there's no way to fix it.  So we should really be validating the json somewhere

Reply:
I'll open a separate PR for this